open-api/rest-catalog-open-api.yaml

```diff
@@ -134,9 +134,17 @@ paths:
     post:
       tags:
         - OAuth2 API
-      summary: Get a token using an OAuth2 flow
+      summary: Get a token using an OAuth2 flow (DEPRECATED for REMOVAL)
+      deprecated: true
       operationId: getToken
       description:
+        The `oauth/tokens` endpoint is **DEPRECATED for REMOVAL**. It is _not_ recommended to
```

Comment:
What does target version mean here? The spec does not really have a general version the last time we discussed.

Reply:
I think the target version should be tied to the reference implementation where the default behaviors will change.

As we discussed before, we can change the requirement that URI be set with the 1.7 release.  However, we can't change the behaviors until 2.0.  So I think we should reference the 2.0 release as the target for removal.

In general, we shouldn't add deprecations that don't clearly indicate a target for removal and reference to the replacement.  In this case we don't have clarity on the exact replacement but I think the current reference to the issue is fine.

Reply:
I'm proposing to include it in 1.6.0 and not wait for 1.7.0 - added a note that mentions "Iceberg (Java) **1.6.0**", because that's the next release this change can realistically go into.

Sure, the improvement proposal document mentioned Iceberg 1.7.0, because at the time of writing it (June), 1.7.0 was the next release the deprecation could go in. 1.6.0 was considered to be released in June ([meeting notes](https://docs.google.com/document/d/1YuGhUdukLP5gGiqCbk0A5_Wifqe2CZWgOd3TbhY3UQg/edit#heading=h.z3dncl7gr8m1)). But now it's already July and 1.6.0 is still not ready to be released.

This change is part of an effort to address a potential security issue. I think that fixes and mitigations for security are a good example for the non-hard rules mentioned in the [Contributing page](https://iceberg.apache.org/contribute/#contributing).

The improvement proposal document assumes 6 months migration period (aka two non-patch Iceberg/Java releases), which is IMO okay considering that this effort is about security.

Reply:
I think Daniel is pointing at when the endpoint will go:

![image](https://github.com/apache/iceberg/assets/1134248/395ca590-0b61-4fe3-b9a9-8cacd19b08e5)

So that should be 1.8 or 2.0 according to the proposal, where it is cleaner to remove it at 2.0.

Reply:
I think it's fine to include the warning in 1.6, but we shouldn't have a behavior change until 2.0 (I don't think a minor version release would be appropriate if we're significantly changing any of the behavior).  I would just update the notices to reference 2.0 for removal.

Reply:
I don't see blocker to have deprecation in 1.6.0 as soon as it doesn't actually change the current behavior (else it's a breaking change).
So I would deprecate (comment) for 1.6.0 and plan the removal in 1.8 or even better 2.0.

Reply:
To calm people down a bit: this change deprecates the endpoint, with the plan to remove the endpoint from the REST specification. That's nothing to do with the REST client implementation.

Reply:
@snazy I don't really understand the pushback here.  When we deprecate something, we need to document the target for removal (this is part of our deprecation process).  While the client implementation is separate from the spec, the two are related in behavior.

To be clear, we need to include messaging on removal before this is ready to commit.

Reply:
The misunderstanding is on my side. I've fixed it, but never pushed the change from my local branch.

Reply:
@danielcweeks I'm confused. I think there's no problem to merge the deprecation flag and message right now, correct ? So, I propose to merge this PR right now, and plan "removal" (impacting client) on 2.0.

Reply:
I think we're good now.  The issue was that the target removal was 1.7 instead of 2.0, which I think was just confusion about deprecation vs. removal.

Reply:
@danielcweeks yes, that's why I was confused 😄 It seems we were "mixing" deprecation and removal. I agree it looks good now. Thanks !